packet/routes/freshmen.py

```diff
@@ -25,11 +26,6 @@ def essays(info=None):
 @before_request
 def submit_essay(info=None):
     formdata = request.form
-    packet = Packet.query.filter_by(freshman_username=info['uid']).first()
-
-    packet.info_eboard = formdata['info_eboard']
-    packet.info_events = formdata['info_events']
-    packet.info_achieve = formdata['info_achieve']
-    db.session.commit()
-
-    return redirect("/essays", 302)
+    if set_requirements(info['uid'], formdata['info_eboard'], formdata['info_events'], formdata['info_achieve']):
+        return redirect("/essays", 302)
+    return redirect("/essays", 500)
```

Comment:
If `set_requirements` fails you're going to get an exception, not a non-true return value. Could you refactor this to better handle those failure cases?